### PR TITLE
fix(TDI-46791): tBigQueryInput - Component returning extra row of NULL values

### PR DESCRIPTION
**Pull request by Tanya5091**
fix for ticket: https://jira.talendforge.org/browse/TDI-46791
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/ServiceAccountBigQueryHelper.javajet [hunk 1]
```diff
@@ -147,7 +147,9 @@ class ServiceAccountBigQueryUtil_<%=cid%> {
                     .flatMap(iterable ->
                             java.util.Optional.ofNullable(java.util.stream.StreamSupport
                                 .stream(iterable.spliterator(), false)
-                                .collect(java.util.stream.Collectors.toList())))
+                    .filter(job -> job.getStatistics().getScriptStatistics() != null)
```
[the remark this answers is not in the document]
Comment (mbasiuk-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/ServiceAccountBigQueryHelper.javajet:150]: Another thing, this expression may be incorrect in this place. In the case of script statistics is null (we won't add it to result), but in OAuth 2 we add it.

File: main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/BigQueryInputQueryHelper.javajet [hunk 2]
```diff
@@ -257,14 +257,10 @@ class BigQueryUtil_<%=cid%> {
...
             %>
-            try {
-                if (!job.getStatistics().getScriptStatistics().getEvaluationKind().toString().equals("EXPRESSION")){
-                    jobList.add(job);
-                }
-            }
-            catch (Exception e){
-                jobList.add(job);
+            if (job.getStatistics().getScriptStatistics() != null && "EXPRESSION".equals(job.getStatistics().getScriptStatistics().getEvaluationKind())){
+                continue;
```
Comment (bgunics-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/BigQueryInputQueryHelper.javajet:261]: If we continue it here then the child statistics won't contain the expression evaluation step. (Not a big deal but I wonder if we could keep it)

Comment (mbasiuk-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/BigQueryInputQueryHelper.javajet:261]: We can move the statistics adding code piece before this if.

File: main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/ServiceAccountBigQueryHelper.javajet [hunk 3]
```diff
@@ -147,6 +147,7 @@ class ServiceAccountBigQueryUtil_<%=cid%> {
                     .flatMap(iterable ->
                             java.util.Optional.ofNullable(java.util.stream.StreamSupport
                                 .stream(iterable.spliterator(), false)
+                                .filter(job -> !"EXPRESSION".equals(job.getStatistics().getScriptStatistics().getEvaluationKind()))
```
Comment (mbasiuk-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/ServiceAccountBigQueryHelper.javajet:150]: It seems this piece of code has to be moved to:
https://github.com/Talend/tdi-studio-se/blob/master/main/plugins/org.talend.designer.components.localprovider/components/tBigQueryInput/tBigQueryInput_begin.javajet#L342

**Review by olhava — APPROVED** [review]

**Pull request merged**